source/server/filter_chain_factory_context_callback.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/api/v2/listener/listener.pb.h"
+#include "envoy/common/pure.h"
+#include "envoy/server/filter_config.h"
+
+namespace Envoy {
+namespace Server {
+
+/**
+ * Handles FilterChainFactoryContext creation. Filter chain contexts are maintained in batch.
```

Comment:
What does "maintained in batch" mean? I'm a little confused about the purpose of this interface and why we need batching per the APIs below. Can you explain more?

Reply:
The idea of batch is derived from the tree-structured-lookup-table. We always use a batch of FilterChainContext to create that table. The de facto batch at this moment of the FilterChainContext is `all the filter chains inside a listener`.

Reply:
obsoleted code